earth_enterprise/src/common/CacheSizeCalculations.h

```diff
@@ -39,9 +38,9 @@ inline uint64 GetObjectSize(const std::vector<T> &vec) {
 inline uint64 GetObjectSize(const std::string &str) {
     return sizeof(str) + (str.capacity() * sizeof(char));
 }
-// determine amount of memory used by a shared_ptr and the object it points to
+// determine amount of memory used by a khRefGuard and the object it points to
 template<class T>
-inline uint64 GetObjectSize(const std::shared_ptr<T> &guard) {
-    return sizeof(guard) + (guard ? guard->GetSize() : 0);
+inline uint64 GetObjectSize(const khRefGuard<T> &guard) {
```

Comment:
Something has gone wrong with your merge. We should be using the `shared_ptr` version of this function, not the `khRefGuard` version.